### ISSUE TITLE
Gradient penalty interpolation is wrong

### ISSUE THREAD

**Issue by iszotic** [LynnHo/CycleGAN-Tensorflow-Simple#21]
Hi, I really like your code.
I found a little detail in line 94
https://github.com/LynnHo/CycleGAN-Tensorflow-2/blob/master/tf2gan/loss.py
the operation should be `inter = b + alpha * (a - b)` no `inter = a + alpha * (b - a)`, doing this fixes the training with gradient penalty

**Comment by LynnHo** [comment]
@iszotic Thank you. Actually, both `inter = b + alpha * (a - b)` and `inter = a + alpha * (b - a)` are correct, since `alpha` is a random number sampled from [0,1].

**Comment by iszotic** [comment]
mmm does the gradient between one and the other computed differently?, I'm not very knowledgeable about backprop but maybe the order changes the gradient.

**Comment by LynnHo** [comment]
@iszotic 
1. `b + alpha * (a - b) = alpha * a + (1 - alpha) * b ` 
2. `a + alpha * (b - a) = (1 - alpha) * a + alpha * b = beta * a + (1 - beta) * b`, here we let `beta = 1 - alpha`

the same thing.
